### ISSUE TITLE
More visible error messages when getting restricted due to import protection

### ISSUE THREAD

**Issue by MikeBellika** [nuxt/nuxt#28717]
### Describe the feature

I just tried to import a value from my server endpoints and was met with a 500 error on the page:
![image](https://github.com/user-attachments/assets/11aad72a-3f4b-4042-a72d-e9e8d1298c36)

This error is a bit misleading and it took me a while to track down that it's happening because I'm importing a value from `~/server`:
```vue
<script setup lang="ts">
import { myArray } from "~/server/api/index";
type MyArrayTypes = (typeof myArray)[number];

defineProps<{ option: MyArrayTypes }>();
</script>
<template>
  <div>
    {{ option }}
    <pre>
      {{ myArray }}
      </pre
    >
  </div>
</template>
```
After restarting the app a couple of times, I now see that I'm getting this error message on startup: `Importing from server is not allowed in the Vue part of your app. [importing server/api/index from components/TestComponent.vue]`.

However this error message only shows up once when starting the app so it's quite easy to overlook. Especially on a larger app, where there might be many warnings/error/logs on startup.

So in conclusion, it would be nice if either/both:
1. The import error shows up continuously. Maybe removing this check? https://github.com/nuxt/nuxt/blob/3c1582b6ff668fa331f61ad535e395276792cee2/packages/nuxt/src/core/plugins/import-protection.ts#L79-L80
2. The error message on the page was more accurate.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).
